### DISCUSSION TITLE
How to copy code by terminal without the extra vertical lines?

### DISCUSSION THREAD

**Discussion by dlintw** [LazyVim/LazyVim#1420] [General]
Here is a example bash code, If I want copy & paste between terminal & browser, it will show the extra vertical lines, how to prevent this?  In, this example, there is no git diff, if exist, there will be more vertical bars. How to setup a toggle key to disable these lines before I copy code.
```
  basic_check() {
  │ for m in "${must[@]}"; do
  │ │ f="$d/$m"
  │ │ if [[ -r $f.list.xz ]]; then
  │ │ │ echo "$f processed"
  │ │ else
  │ │ │ if ! [[ -r $f.tar.gz ]]; then
  │ │ │ │ echo "Err:$LINENO missing $f.tar.gz"
  │ │ │ │ exit 1
  │ │ │ fi
  │ │ fi
  │ done
  │ echo "Done basic file check"
  }

```

**Comment by abeldekat** [comment]
Hello @dlintw,

Those lines are rendered by the following plugins:

- [indent-blankline.nvim](https://github.com/lukas-reineke/indent-blankline.nvim)
- [mini.indentscope](https://github.com/echasnovski/mini.indentscope)

The lines are "virtual" and *not* included in a copy.
You can verify this by copying your code to another file. Then, open the file with `nvim -u NORC file_containing_copied_basic_check`

Toggling is possible, for example:
`vim.g.miniindentscope_disable = not vim.g.miniindentscope_disable`
`vim.g.indent_blankline_enabled = not vim.g.indent_blankline_enabled`

[mini disabling recipes](https://github.com/echasnovski/mini.nvim/blob/3a3e19d7b7850e711485b8adde73ba44789f2cf6/doc/mini.txt#L270)
[indent_blankline enabled](https://github.com/lukas-reineke/indent-blankline.nvim/blob/9637670896b68805430e2f72cf5d16be5b97a22a/doc/indent_blankline.txt#L347)

**Comment by dlintw** [comment]
I copy & paste from terminal(putty), After 

```
:let g:miniindentscope_disable = v:false
:let g:indent_blankline_enabled = v:false

```

Now, here is the result, there contains 
1. git diff bar
2. tab replacement chars '>'
3. cursor block vertical lines
 
How could I setup a <leader> hotkey to toggle these settings?  If it possible, wish this can default release of lazyvim for toggle function.

```
▎ for y in "$S"/????; do
▎ > for d_dir in "$y"/????????; do
▎ > > d=$(basename "$d_dir")
▎ > > if [[ $d -ge $from_date && $d -le $to_date ]]; then
▎ > > │ echo "test_date=$d"
▎ > > │ mkdir -p "$d"  <- I put my cursor here
▎ > > │ rm -rf "$d"
▎ > > │ cp -a -s "$d_dir" "$d"
▎ > > │ ./test_recover.sh "$d"
▎ > > │ mkdir -p "$target_dir/${d:0:4}"
▎ > > │ rm -rf "${target_dir:?}/${d:0:4}/$d"
▎ > > │ mv "$d" "$target_dir/${d:0:4}/$d"
▎ > > │ break
▎ > > fi
▎ > done
▎ done

```

**Comment by abeldekat** [comment]
> If I want copy & paste, it will show the extra vertical lines

I misunderstood the "by terminal" part in the header of your question, my bad.

Unfortunately, I don't use putty.
Nevertheless, allow me to ask why you want to copy by terminal instead of using vim's yank? In LazyVim, you can for example visually select the lines and press `y` to copy them to the system clipboard.

**Reply by dlintw** [reply]
1. In Windows OS, I don't know such function workable?
2. Even in Linux, terminal, between browser, it is also cause this issue.
3. I use putty->tmux -> vim, I don't know how to proper setup it workable just by 'y' command.

**Reply by dpetka2001** [reply]
> 2. Even in Linux, terminal, between browser, it is also cause this issue.

In linux, there shouldn't be a problem for copying between browser and terminal. That is as long as you have a proper provider installed for the clipboard integration (You could check in your Neovim with the command `:checkhealth provider` and look what it says for the `Clipboard`, in case the results are folded you can unfold the lines with `zR`). Please take a look at Neovim [docs](https://neovim.io/doc/user/provider.html#provider-clipboard) for available clipboard providers and choose one according to your Operating System. Unfortunately, I cannot speak for the putty part as I also don't use it myself. But for Linux terminal, browser and even tmux it should just work provided you have an appropriate clipboard provider installed.

**Reply by abeldekat** [reply]
Thanks @dpetka2001!

I was just thinking of asking for your help...) Regarding `linux`, I have nothing to add.

Maybe, the [zen-mode](https://github.com/folke/zen-mode.nvim) plugin can be of use.

Less helpful, mentioned for completeness:
* does putty have a `cat` command? That will output the file as is in the terminal.
* the `scp` command.

**Reply by dpetka2001** [reply]
As far as putty is concerned, it all depends if it uses the system clipboard of the Operating System. If it does, then there shouldn't be any problems. But I don't use it, so I can't say for sure. After some googling, I may have found a relevant [topic](https://unix.stackexchange.com/questions/623762/copy-from-vim-putty-session-to-windows-clipboard/628655#628655) that might be of help, but I cannot attest to its correctness. It seems there are options in putty configuration to use the system clipboard, but I don't know if that's the default behavior, so you might want to check that out @dlintw. 

@abeldekat I'm happy to answer and provide help given that the problem is something within my understanding or at least relevant to my knowledge, so that I don't provide misinformation :smile:

**Reply by dlintw** [reply]
for my windows + putty environment + no gui linux server, follow the steps, I can copy code from linux to windows by 'y' command,

1. install server on windows 
https://github.com/lemonade-command/lemonade
2. install client on linux
3. setup putty port forwarding R2489:127.0.0.1:2489

**Comment by dlintw** [comment]
No one knows which plugin cause the extra vertical lines?  

1. git diff bar
2. tab replacement chars '>'
3. cursor block vertical lines
